### DISCUSSION TITLE
Method Illuminate\Database\Eloquent\Collection::delete does not exist.

### DISCUSSION THREAD

**Discussion by yaquawa** [laravel/framework#23985] [General] [answered]
- Laravel Version: 5.6
- PHP Version: 7.2
- Database Driver & Version: mysql

### Description:

Providing a `delete` method to eloquent collection would be very helpful.

```php
User::where('in_black_list', true)->get()->delete();
```

**Comment by sisve** [comment]
`User::where(...)` doesn't return a Collection, but a Eloquent query builder. Is this really the entire code snippet required to reproduce the problem?

**Comment by yaquawa** [comment]
@sisve Sorry! I forgot the `get()` method. I've edited the example code.

**Comment by bonzai** [comment, marked as answer]
Just use `->each()`:

```php
User::where('in_black_list', true)->get()->each->delete();
```

https://laravel.com/docs/5.6/collections#higher-order-messages

**Comment by ntzm** [comment]
Or just `User::where('in_black_list', true)->delete()`

**Comment by JoseCage** [comment]
> ```
> User::where('in_black_list', true)->get()->each->delete();
> ```

Helpful.

Thanks for your help.
